src/Handler/TagHandler.php

```diff
@@ -40,18 +45,20 @@ class TagHandler
     /**
      * Constructor
      *
-     * @param CacheInvalidator $invalidator The invalidator instance.
-     * @param string           $tagsHeader  Header to use for tags, defaults to X-Cache-Tags.
+     * @param CacheInvalidator $invalidator  The invalidator instance.
+     * @param string           $tagsHeader   Header to use for tags, defaults to X-Cache-Tags.
+     * @param int              $headerLength Maximum header size in bytes, defaults to 7500.
      *
      * @throws UnsupportedProxyOperationException If CacheInvalidator does not support invalidate requests
      */
-    public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cache-Tags')
+    public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cache-Tags', $headerLength = 7500)
```

Comment:
we should probably switch to an OptionsResolver for tags header and header length. more expressive and more flexible for future configuration options. we would have `array $options = []` here and then something like

```
$resolver = new OptionsResolver();
$resolver->setDefaults([
    'tags_header' = 'X-Cache-Tags',
    'header_length' = 7500,
]);
$resolver->setAllowedType('tag_header', 'string');
$resolver->setAllowedType('header_length', 'int');

$options = $resolver->resolve($options);
$this->tagsHeader = $options['tags_header'];
$this->headerLength = $options['header_length'];
```